docs/api-tutorials/custom-reporter.md

```diff
@@ -0,0 +1,113 @@
+
+| Constant                 | Event Name  | Event Arguments | Description                                                                                 |
+| ------------------------ | ----------- | --------------- | ------------------------------------------------------------------------------------------- |
+| `RUNNER_EVENT_END`       | `end`       | _(n/a)_         | All [Suite]s, [Test]s and [Hook]s have completed execution                                  |
+| `RUNNER_EVENT_FAIL`      | `fail`      | `Test`, `Error` | A [Test] has failed or thrown an exception                                                  |
+| `RUNNER_EVENT_HOOK`      | `hook`      | `Hook`          | A [Hook] is about to execute                                                                |
+| `RUNNER_EVENT_HOOK_END`  | `hook end`  | `Hook`          | A [Hook] has completed execution                                                            |
+| `RUNNER_EVENT_PASS`      | `pass`      | `Test`          | A [Test] has passed                                                                         |
+| `RUNNER_EVENT_PENDING`   | `pending`   | `Test`          | A [Test] was skipped                                                                        |
+| `RUNNER_EVENT_RETRY`     | `retry`     | `Test`, `Error` | A [Test] failed, but is about to be retried; never emitted unless `retry` option is nonzero |
+| `RUNNER_EVENT_BEGIN`     | `start`     | _(n/a)_         | Execution will begin                                                                        |
+| `RUNNER_EVENT_SUITE`     | `suite`     | `Suite`         | The [Hook]s and [Test]s within a [Suite] are about to be executed                           |
+| `RUNNER_EVENT_SUITE_END` | `suite end` | `Suite`         | The [Hook]s and [Test]s within a [Suite] (and any children [Suite]s) completed execution    |
+| `RUNNER_EVENT_TEST`      | `test`      | `Test`          | A [Test] is about to be executed                                                            |
+| `RUNNER_EVENT_TEST_END`  | `test end`  | `Test`          | A [Test] has completed execution                                                            |
+| `RUNNER_EVENT_WAITING`   | `waiting`   | _(n/a)_         | Waiting for `global.run()` to be called; only emitted when `delay` option is `true`         |
+
+**Please use these constants** instead of the event names in your own reporter! This will ensure compatibility with future versions of Mocha.
+
+> It's important to understand that all suite callbacks will be run _before_ the [Runner] emits `RUNNER_EVENT_BEGIN`. Hooks and tests, however, won't run until _after_ the [Runner] emits `RUNNER_EVENT_BEGIN`.
+
+## Custom Reporter for Browser
+
+As of Mocha v6.0.0, custom reporters are _only_ "officially" supported in Node.js. [Mochify](https://npm.im/mochify) might help you. YMMV!
+
+[runner]: /api/mocha.runner
+[test]: /api/mocha.test
+[hook]: /api/mocha.hook
+[suite]: /api/mocha.suite
```

Comment:
Why was the [Wiki page](https://github.com/mochajs/mocha/wiki/Third-party-reporters) poo-poo'd in favor of this?

Reply:
I would like to consolidate tutorials on the documentation site.  It is painful to have to go to separate sites for information.

I am not thrilled about how the tutorial is buried in the navigation, but #3663 should allow us to move the "Tutorials" nav section to the top.